packages/grafana-ui/src/components/PieChart/PieChart.tsx

```diff
@@ -153,18 +157,51 @@ export const PieChartSvg: FC<SvgProps> = ({
   };
 
   const showLabel = displayLabels.length > 0;
-  const total = values.reduce((acc, item) => item.numeric + acc, 0);
+  const total = fieldDisplayValues.reduce((acc, item) => item.display.numeric + acc, 0);
   const layout = getPieLayout(width, height, pieType);
 
+  const getPieSlice = (
```

Comment:
This feels like a react component to me. What about having a PieSlice component that would handle data links in it rather than this render function in a component? From PieChartSvg render POV it would look like this:
```return pie.arcs.map(arc => <PieSlice .../>)```